src/features/keymap/notifier.rs

```diff
@@ -0,0 +1,67 @@
+use crate::communication;
+use crate::error::*;
+use crate::wrapper::channel;
+use crate::wrapper::thread;
+use xcb;
+use xcb::xkb;
+
+pub(super) struct Notifier {
+    id: usize,
+    sender: channel::Sender<communication::Message>,
+}
+
+impl Notifier {
+    pub(super) const fn new(id: usize, sender: channel::Sender<communication::Message>) -> Self {
+        Self { id, sender }
+    }
+}
+
+impl thread::Runnable for Notifier {
+    fn run(&self) -> Result<()> {
```

Comment:
To be honest, I don't really understand what is going on here. Wrapping all xcb code into a wrapper module with descriptive method names should help, but maybe sprinkle in some comments :)